tests/integration/tokenomics/token_logic_modules/tlm_processor_suite_test.go

```diff
@@ -0,0 +1,172 @@
+package token_logic_modules
+
+import (
+	"context"
+	"math"
+	"testing"
+
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/pokt-network/poktroll/app/volatile"
+	"github.com/pokt-network/poktroll/cmd/poktrolld/cmd"
+	testkeeper "github.com/pokt-network/poktroll/testutil/keeper"
+	"github.com/pokt-network/poktroll/testutil/proof"
+	"github.com/pokt-network/poktroll/testutil/sample"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
+	tlm "github.com/pokt-network/poktroll/x/tokenomics/token_logic_module"
+)
+
+type tlmProcessorTestSuite struct {
+	suite.Suite
+
+	ctx     context.Context
+	keepers testkeeper.TokenomicsModuleKeepers
+
+	service  *sharedtypes.Service
+	app      *apptypes.Application
+	supplier *sharedtypes.Supplier
+
+	proposerConsAddr cosmostypes.ConsAddress
+	sourceOwnerBech32,
+	foundationBech32 string
```

Comment:
Could we use `dao` instead of `foundation` for consistency with how it's used in non-testing code?